ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js

```diff
@@ -509,6 +519,41 @@ export default class ConfirmApproveContent extends Component {
     return titleTokenDescription;
   }
 
+  renderTitle() {
+    const { t } = this.context;
+    const { isSetApproveForAll, setApproveForAllArg } = this.props;
+    const titleTokenDescription = this.getTitleTokenDescription();
+
+    let title = t('allowSpendToken', [titleTokenDescription]);
+
+    if (isSetApproveForAll) {
+      title = t('approveAllTokensTitle', [titleTokenDescription]);
+      if (setApproveForAllArg === false) {
+        title = t('revokeAllTokensTitle', [titleTokenDescription]);
+      }
+    }
+    return title;
```

Comment:
comment & no need to change:

Another way to go about this is 
```
let title;
// ... 
return title || t('allowSpendToken', [titleTokenDescription]);
```

could save 1 translation call